services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/ApiBlockingFilterActuatorITests.kt

```diff
@@ -0,0 +1,27 @@
+package fi.hel.haitaton.hanke
+
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.security.test.context.support.WithAnonymousUser
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+@SpringBootTest(
+    properties = ["haitaton.api.disabled=true", "management.server.port="],
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
```

Comment:
What does this do? We have dropped this from other integration tests.

Reply:
This will start the servlet engine in a random free port. Without using it I could not get these test with Actuator endpoints and RestTemplate to work. Also, it makes using `@LocalServerPort` to work and via it we have the actual port where the calls should be pointed at.

Setting `management.server.port=` is also necessary and without it the call gives an 404 error for Actuator endpoints and after that 401 error for trying to access `/error` without authorization.

Also, now I can test with this same live-like test the `/status` endpoint as well.

Reply:
I changed it. Thank you for your help. I tried a lot of things but missed this one.